### PR TITLE
[SPARK-39723][R] Implement functionExists/getFunc in SparkR for 3L namespace

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
1, implement functionExists/getFunc in SparkR
2, update doc of ListFunctions

### Why are the changes needed?
for 3L namespace

### Does this PR introduce _any_ user-facing change?
yes, new API functionExists

### How was this patch tested?
added UT

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
LGTM, pending tests.

**Comment by zhengruifeng** [comment]
an irrelevent test `[Run / Build modules: catalyst, hive-thriftserver](https://github.com/zhengruifeng/spark/runs/7281537061?check_suite_focus=true#logs)` got stuck for more than 3hours, let me retest it.

File: R/pkg/tests/fulltests/test_context.R [hunk 1]
```diff
@@ -21,10 +21,11 @@ test_that("Check masked functions", {
   # Check that we are not masking any new function from base, stats, testthat unexpectedly
   # NOTE: We should avoid adding entries to *namesOfMaskedCompletely* as masked functions make it
   # hard for users to use base R functions. Please check when in doubt.
-  namesOfMaskedCompletely <- c("cov", "filter", "sample", "not")
+  namesOfMaskedCompletely <- c("cov", "filter", "sample", "not", "getFunction")
```
Comment (zhengruifeng) [inline R/pkg/tests/fulltests/test_context.R:24]: to pass `test_context`, I need to mask the build-in [`getFunction`](https://www.rdocumentation.org/packages/methods/versions/3.6.2/topics/methodUtilities) , is this OK? @HyukjinKwon

Comment (HyukjinKwon) [reply R/pkg/tests/fulltests/test_context.R:24]: Should probably we should define this with `setMethod` (see `functions.R`) to avoid the complete masking.

Comment (HyukjinKwon) [reply R/pkg/tests/fulltests/test_context.R:24]: Ah .. even with that the signature would likely conflict. hm

Comment (HyukjinKwon) [reply R/pkg/tests/fulltests/test_context.R:24]: Let's use a different name for now. e.g., `getFunc` and explain it in the docs.

Comment (zhengruifeng) [reply R/pkg/tests/fulltests/test_context.R:24]: got it, thanks. let me update the pr.

Comment (HyukjinKwon) [reply R/pkg/tests/fulltests/test_context.R:24]: cc @falaki @felixcheung @shivaram FYI

File: R/pkg/pkgdown/_pkgdown_template.yml [hunk 2]
```diff
@@ -266,7 +266,9 @@ reference:
   - databaseExists
   - dropTempTable
   - dropTempView
+  - functionExists
   - getDatabase
+  - getFunction
```
Comment (HyukjinKwon) [inline R/pkg/pkgdown/_pkgdown_template.yml:271]:
```suggestion
  - getFunc
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
Merged to master, thanks @HyukjinKwon

**Pull request closed without merging**